src/shared_realm.cpp

```diff
@@ -554,5 +555,110 @@ util::Optional<int> Realm::file_format_upgraded_from_version() const
     return util::none;
 }
 
+#define VERSION_CAST(version_id) {(version_id).version, (version_id).index}
```

Comment:
It doesn't cast to either type, just gives the braced initializer list representation.